retrieval/scrape.go

```diff
@@ -418,13 +386,19 @@ func (sl *scrapeLoop) run(interval, timeout time.Duration, errc chan<- error) {
 			}
 
 			samples, err := sl.scraper.scrape(scrapeCtx, start)
+			app := sl.batchAppender.StartBatch()
```

Comment:
StartBatch creates the watermark as "now" again. It should get `start` (from the previous line) passed in here so that it aligns with the timestamp the samples will get. (Obviously, if samples in the scrape set their own timestamp, all is lost anyway.)